packages/babel-helper-remap-async-to-generator/src/index.js

```diff
@@ -145,6 +145,15 @@ function plainFunction(path: NodePath, callId: Object) {
     ]);
     declar._blockHoist = true;
 
+    if (path.parentPath.isExportDefaultDeclaration()) {
+      // change the path type so that replaceWith() does not wrap
+      // the identifier into an expressionStatement
+      path.type = "FunctionExpression";
+      path.parentPath.insertBefore(declar);
+      path.replaceWith(t.identifier(asyncFnId.name));
```

Comment:
This isn't quite correct, so I have another way we should try, but it may also cause other problems, if it doesn't work I'm fine with doing this as-is.

Since it is a function declaration, it should technically be exported up front, but since this is being replaced with an expression, the export won't be exposed right away. So instead of

    export default asyncId;

we should do

    export {asyncId as default};

Can we try that and see if it introduces any issues?